canonical-web-and-design/maas-ui
# Convert component scss into mixins + import all styles into index.scss file.

## Done
- Converted all component scss into `_index.scss`'s (for nicer import paths) and removed them from being directly imported into React components.
- Wrapped all "global" styles in mixins.
- Created a top-level `index.scss` file where all scss in the project is listed. I think this is nicer because now we only have to import vanilla+settings only once, instead of as needed per-component (which I believe was also causing some settings overrides).
- Upgraded to Vanilla 2.8.0 and fixed regressions + removed overrides (mainly to do with contextual menu positioning and form spacing).
- Removed some "easy win" redundant styles that were either fixed in the latest Vanilla or copied unnecessarily from legacy, but we should definitely go over what we have more thoroughly.

## QA
- Click around everywhere in the React pages and check that styling still looks ok everywhere.
- See if you agree with the folder structure

## Fixes
Fixes #747
